ggml-cuda.cu

```diff
@@ -11627,6 +11517,31 @@ GGML_CALL void ggml_backend_cuda_get_device_memory(int device, size_t * free, si
     CUDA_CHECK(cudaMemGetInfo(free, total));
 }
 
+GGML_CALL bool ggml_backend_cuda_register_host_buffer(void * buffer, size_t size) {
+    if (getenv("GGML_CUDA_NO_PINNED") != nullptr) {
```

Comment:
Are there plans to change this from an environment variable before merging?

Reply:
It's the same environment variable that was already used in `ggml_cuda_host_malloc`. I am not sure if it is worth making this more complicated, pinning the buffer should be fairly safe. It may cause issues in systems with very low RAM because it will lock the memory and make it unavailable to other applications (pinned memory cannot be swapped), but that's not entirely unexpected if you try to run a model without enough memory.